plugin/trino-clickhouse/src/test/java/io/trino/plugin/clickhouse/TestClickHouseConnectorTest.java

```diff
@@ -901,6 +929,55 @@ public void testTextualPredicatePushdown()
         assertThat(query("SELECT regionkey, nationkey, name FROM nation WHERE name = 'romania'"))
                 .returnsEmptyResult()
                 .isFullyPushedDown();
+        assertThat(query("SELECT regionkey, nationkey, name FROM nation WHERE name = 'romania' OR comment = 'P'"))
+                .returnsEmptyResult()
+                .isFullyPushedDown();
+
+        String withConnectorExpression = " OR some_column = 'x'";
+        try (TestTable table = new TestTable(
+                onRemoteDatabase(),
+                "tpch.test_textual_predicate_pushdown",
+                """
+                        (
+                        some_column String,
+                        a_string String,
+                        a_string_alias text,
+                        a_fixed_string FixedString(1),
+                        a_string_nullable Nullable(String),
+                        a_string_alias_nullable Nullable(text),
+                        a_fixed_string_nullable Nullable(FixedString(1)),
+                        a_enum_1 Enum('hello', 'world', 'a', 'b', 'c'),
+                        a_enum_2 Enum('hello', 'world', 'a', 'b', 'c'))
```

Comment:
Sure we can @Praveen2112 

Appropriate testing of pushdown support is a O(m*n*k) complexity:
m = Operations, e.g equal (=), lessThan(<=), etc., actually any simplest operation which could be mapped as a row in connectorExpressionRewriter
n = types, all types from TypeMapping for a connector 
k = data, any magic data could be inserted within type, e.g. https://github.com/trinodb/trino/pull/23496

Reply:
I do agree it comes with a cost of additional CPU cycles - The reason for this special type is that - `Lowcardinality` is similar to enum and we need to ensure things don't break in the future with newer version of clickhouse - Hope it could save some additional CPU cycles when debugging - to compensate this cost.